Quicktest pentodes

colincci opened the pull request:
Hi Richard

I have created a quicktest-pentode function modelled after your quicktest-triode function.  It is now working and appears to be giving me nearly identical results to uTracer windows GUI.  I have it reporting in the comments of the log file the same parameters which the windows GUI outputs for pentode quicktests.

I have used the Va_Meas and Vs_Meas values to do the final calculations for the key parameters.  I notice that the quicktest-triode is using the nominal Va and Vs values.  Have you done any experimentation on this?  I am not sure which is 'correct' or what Ronald is doing in the windows GUI.

The next thing I think I will tackle is the matching database and run scripts for quicktesting and matching pentodes.  

Thanks

Colin

Commit:
Begin working on spliting out Constants and Config,Cal data

colincci committed:
Configure file and cal file tested and working.
Updated the POD usage.
Added 6201 tube type.
Started working on El84 tube type configuration.
Pentode scans not yet working.

Commit:
Add in the get_cal() subroutine to read the app.cal file

colincci committed:
EL84 Pentode Quicktests are now working and giving nearly identical results to the uTracer windows GUI.

colincci committed:
Found a bug where the cal data was read from app.cal in tracer.pl
but from app.ini in uTracerConstants.pm.
Have removed the calibration data from app.ini and moved the
get_cal function from tracer.pl to uTracerConstants.
The app.cal file now is read in by uTracerConstants.pm and presented
to tracer as the $cal calibration hash reference.

warewolf commented:
> I have used the Va_Meas and Vs_Meas values to do the final calculations for the key parameters. I notice that the quicktest-triode is using the nominal Va and Vs values. Have you done any experimentation on this? I am not sure which is 'correct' or what Ronald is doing in the windows GUI.

I think I selected the nominal (e.g. Va and Vs "should" be these values, but due to hardware stuffs they don't) because it made pulling the results out of an array of results easier, because I could say "I expect Va to be voltage X, go find voltage X in the results".  At least that was the idea, from trying to re-implement stuff from https://wtfamps.wordpress.com/mugmrp/ -- but now that I look back at the code, I set the "todo" list of values to measure, then simply say "find me the highest and lowest values", so yeah, I totally can use the measured values not the nominal ones.

P.s: dude, thank you for the pentode quicktest.  I suck at math, and it took months for me to find an explanation of the 5 measurement points for a triode quicktest that I understood.  I never took advanced arithmetic in school, so deltas were *completely* foreign to me.

warewolf commented:
Oh, my fiancee is going to be home from being away for quite some time, so my schedule this weekend of coding turned into cleaning the house!  Apologies for not getting to code integrated.  There's a couple (minor) things I want to do slightly differently.

colincci committed:
-Added TubeDatabase.pm module
-Added --store option to automatically store the quicktest data in the
sqlite database.
-Added pentodes, pentode_types and triode_types table to the sqlite
schema
-Added run_test-pentode.sh.  This runs the full quick test, and then
gens the png files, creates the pdf, moves the csv's and pdfs into
the ./pentodes directory and then cleans up after itself.
-Added 6AK5 pentode type to app.ini
-Added dump_db.sh to dump the pentodes table in sqlite to csv file

colincci commented:
Hey no worries about the delay.  I appreciate all the great work you did to get this code to were it is.  I have a strong math background, but it's not like riding a bike.  Use it or loose it!!  It took me a while to wrap my head around the pentode quick test.  And all your magical perl map functions, which are pretty cool.
I don't mean to be inundating you with pull requests and new code, but I had a bit of time free this weekend, which is kind of a miracle and I have a LOTTTT of pentodes I want to get testing with my new uTracer.  But, I want to get the data all sorted out the way I want it before I spend the time doing the actual testing.

colincci committed:
Added 12by7a Tube Type

colincci commented:
Hi Richard
I merged this branch back into my cdc main branch.  I will close this pull request.  You can just pull the updates from the cdc branch and not have to deal with multiple branches.  As I am adding new features, I will branch from my cdc branch and merge them back in when the feature is ready.  So, in future, you should only see pull requests from the cdc branch, unless I have a question about something I am working on in the feature branch.
Colin

warewolf commented:
Saw your e-mail while I was cleaning the house - this is fine; and entirely the reason why I published my code on github!  Keep being awesome.  I can pull stuff in later, but it may end up being a git nightmare for you if our codebases diverge too much.  I'll see if I can get to this later this weekend.
